keps/sig-node/2400-node-swap/README.md

```diff
@@ -181,16 +192,59 @@ By doing so, we can ensure a thorough understanding of the feature's performance
 
 Allocate the swap limit equal to the requested memory for each container and adjust the proportion of swap based on the total swap memory available.
 
-#### Set Aside Swap for System Critical Daemon
+#### Set Aside Swap for System Critical Daemons
+
+**Note** In Beta2, we found that having system critical daemons swapping memory could cause degration of services.
 
-System critical daemons (such as Kubelet) are essential for node health. Usually, an appropriate portion of system resources (e.g., memory, CPU) is reserved as system reserved. However, swap doesn't inherently support reserving a portion out of the total available. For instance, in the case of memory, we set `memory.min` on the node-level cgroup to ensure an adequate amount of memory is set aside, away from the pods, and for system critical daemons. But there is no equivalent for swap; i.e., no `memory.swap.min` is supported in the kernel. 
+System critical daemons (such as Kubelet) are essential for node health. Usually, an appropriate portion of system resources (e.g., memory, CPU) is reserved as system reserved. However, swap doesn't inherently support reserving a portion out of the total available. For instance, in the case of memory, we set `memory.min` on the node-level cgroup to ensure an adequate amount of memory is set aside, away from the pods, and for system critical daemons. But there is no equivalent for swap; i.e., no `memory.swap.min` is supported in the kernel.
 
 Since this proposal advocates enabling swap only for the Burstable QoS pods, this can be done by setting `memory.swap.max` on the cgroups used by the Burstable QoS pods. The value of this `memory.swap.max` can be calculated by:
 
 memory.swap.max = total swap memory available on the system - system reserve (memory)
 
 This is the total amount of swap available for all the Burstable QoS pods; let's call it `TotalPodsSwapAvailable`. This will ensure that the system critical daemons will have access to the swap at least equal to the system reserved memory. This will indirectly act as having support for swap in system reserved.
 
+#### Best Practices
+
+This section is a recommendation for how to set up your nodes with swap if using this feature.
+
+##### Disable swap for system critical daemons
+
+As we were testing this feature, we found degration of services if you allow system critical daemons to swap.
+This could mean that kubelet is performing slower than normal so if you experience this,
+we recommend setting the cgroup for the system slice to avoid swap (ie `memory.swap.max 0`).
```

Comment:
Should we, SIG Node, be more assertive on this? Instead of recommendation, should we disable swap for those critical daemons from builtin?

Reply:
I mentioned this in an sig meeting and I think the thought was that we can’t really enforce disabling system slice as that impacts more than just kubernetes. It could be worrying as I don’t think kubelet has examples of modifying the system slice. It could also impact other users machines in unexpected ways (unclear if we can really say what users are putting in system.slice)

Reply:
We did it long time back, but removed those over the time. I think you are right since we cannot really control what's in system.slice from the open source perspective.